### PR TITLE
Merge ES2K changes

### PR DESCRIPTION
**Pull request by ffoulkes**

### CODE REVIEW

File: docs/simple_l3_mirroring_wcm.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+are present. It differs from the existing Match-Action table in the sense that, it doesn't have any action associated with it. 
+Ideally this table can be configured with Exact or Ternary lookup keys. 
+This new table is defined as a user defined extern object type for the PNA architecture. The corresponding .proto changes that are 
+required to support this new table construct are introduced in p4runtime submodule, and the same needs to be open-sourced. 
+
+## SDE reference for p4 and README_OVS
+p4 program 
+https://github.com/intel-innersource/networking.ethernet.acceleration.vswitch.p4-sde.p4-driver/tree/main/mev_reference_p4_files/simple_l3_mirror_wcm
+
+README_OVS
```
Comment (nupuruttarwar) [inline docs/simple_l3_mirroring_wcm.md:18]: References to innersource

File: docs/simple_l3_mirroring_wcm.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+### Install Wheel Package and p4Runtime files using p4runtime-1.3.0-py3-none-any.whl
+a) uninstall existing p4runtime
+b) Install wheel package
+   #pip install wheel
+c) install p4Runtime files using "p4runtime-1.3.0-py3-none-any.whl" wheel file
+   #pip install p4runtime-1.3.0-py3-none-any.whl
+	
+### Pipeline build procedure:
+a) Handcraft the 'simple_l3_mirror_wcm.conf' file by following the sample conffile 'mev_sample.conf' provided in the '/mev_reference_p4_files' directory.
```
Comment (nupuruttarwar) [inline docs/simple_l3_mirroring_wcm.md:31]: References to file in P4SDE

File: docs/guides/es2k-guide.md [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+
+```bash
+For Fedora distro: yum install libatomic libnl3-devel openssl-devel
+For Ubuntu distro: apt install libatomic1 libnl-route-3-dev libssl-dev
+pip3 install -r requirements.txt
+```
+
+### Install P4 SDE
+
+Obtain a copy of the SDE for the Intel ES2000 IPU and install it on your
```
Comment (nupuruttarwar) [inline docs/guides/es2k-guide.md:20]: ES2100

File: docs/guides/es2k-guide.md [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+
+#### Create IPDF netdevs
+
+After installing ATE Kernel on HOST machine, install the following drivers
+to bind the network devices (netdevs) to the ES2000 target.
+
+```bash
+modprobe idpf
+modprobe auxiliary
+modprobe iecm
```
Comment (nupuruttarwar) [inline docs/guides/es2k-guide.md:119]: iecm module has been removed. Please remove line 119

File: docs/simple_l3_mirroring_wcm.md [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+./install/bin/p4rt-ctl add-entry bro my_control.i_fwd "hdrs.mac[vmeta.common.depth].da=0x000000000940,action=my_control.send(17)"
+./install/bin/p4rt-ctl add-entry bro my_control.l3_l4_match "hdrs.ipv4[vmeta.common.depth].src_ip=192.168.1.0/255.255.255.0,hdrs.ipv4[vmeta.common.depth].dst_ip=192.168.2.0/255.255.255.0,priority=1,action=my_control.mirror_and_send(18)"
+
+Delete rule:
+-----------
+./install/bin/p4rt-ctl del-lut-entry br0 my_control.mir_prof "mir_prof_key=18"
+./install/bin/p4rt-ctl del-entry bro my_control.i_fwd "hdrs.mac[vmeta.common.depth].da=0x000000000940"
+./install/bin/p4rt-ctl del-entry bro my_control.l3_l4_match "hdrs.ipv4[vmeta.common.depth].src_ip=192.168.1.0/255.255.255.0,hdrs.ipv4[vmeta.common.depth].dst_ip=192.168.2.0/255.255.255.0,priority=1"
+
+#### Test traffic from link partner to ES2000
```
Comment (nupuruttarwar) [inline docs/simple_l3_mirroring_wcm.md:55]: ES2100

File: CMakeLists.txt [hunk 6]
```diff
@@ -49,19 +56,27 @@ file(MAKE_DIRECTORY ${PB_OUT_DIR})
 
 option(DPDK_TARGET      "Build for DPDK target" OFF)
 option(TOFINO_TARGET    "Build for Tofino target" OFF)
+option(ES2K_TARGET      "Build for ES2000 target" OFF)
```
Comment (nupuruttarwar) [inline CMakeLists.txt:59]: ES2100

File: cmake/es2k-driver.cmake [hunk 7]
```diff
@@ -0,0 +1,134 @@
+# Definitions for the ES2000 P4 driver package.
```
Comment (nupuruttarwar) [inline cmake/es2k-driver.cmake:1]: ES2100

File: docs/guides/es2k-guide.md [hunk 8]
```diff
@@ -0,0 +1,199 @@
...
+```bash
+source ./scripts/setup_env.sh $IPDK_RECIPE $SDE_INSTALL $DEPEND_INSTALL
+./scripts/copy_config_files.sh $IPDK_RECIPE $SDE_INSTALL
+alias sudo='sudo PATH="$PATH" HOME="$HOME" LD_LIBRARY_PATH="$LD_LIBRARY_PATH" SDE_INSTALL="$SDE_INSTALL"'
+```
+
+#### Create IPDF netdevs
+
+After installing ATE Kernel on HOST machine, install the following drivers
+to bind the network devices (netdevs) to the ES2000 target.
```
Comment (nupuruttarwar) [inline docs/guides/es2k-guide.md:114]: ES2100

File: docs/guides/es2k-guide.md [hunk 9]
```diff
@@ -0,0 +1,199 @@
...
+sudo  ./install/bin/p4rt-ctl add-entry br0 MainControlImpl.l3_l4_match_rx \
+    "hdrs.ipv4[vmeta.common.depth].protocol=0x11,vmeta.common.port_id=0,istd.direction=0,
+    hdrs.ipv4[vmeta.common.depth].src_ip="192.168.1.10",hdrs.ipv4[vmeta.common.depth].dst_ip="192.168.1.20",
+    hdrs.udp[vmeta.common.depth].sport=1000,hdrs.udp[vmeta.common.depth].dport=2000,action=MainControlImpl.send(17)"
+```
+
+ *Note*: See [p4rt-ctl Readme](../clients/p4rt-ctl.rst) for more information
+ on the `p4rt-ctl` utility.
+
+#### Test traffic from link partner to ES2000
```
Comment (nupuruttarwar) [inline docs/guides/es2k-guide.md:191]: ES2100

File: docs/guides/es2k-guide.md [hunk 10]
```diff
@@ -0,0 +1,199 @@
...
+
+Send a packet from a physical port of link partner to the ES2000 using any
```
Comment (nupuruttarwar) [inline docs/guides/es2k-guide.md:193]: ES2100

File: docs/simple_l3_mirroring_wcm.md (hunk 1 above)
Comment (ffoulkes) [reply docs/simple_l3_mirroring_wcm.md:18]: I'm going to remove the document. It's for a P4 file that is not part of the networking recipe.

**Comment by nupuruttarwar** [comment]
Hi Derek, Marketing team confirmed that it's E2100 now no 'S'

**Comment by ffoulkes** [comment]
> Hi Derek, Marketing team confirmed that it's E2100 now no 'S'

I'm inclined to leave things as they are for the initial merge and clean them up in a subsequent pass. We need to talk about the scope of the change, agree on what we should and should not change, and advise the development team and our users of what we are doing.

ES2K is baked into the code and its interfaces. I don't think we're going to want to change `--target es2k` to `--target e2k`, nor do I think we're going to want to rename all the `es2k_` files and `Es2k` classes and the `ES2K_TARGET` conditional. ES2K is the _software_ identifier for this class of targets, and E2100 (or E2000, or E-my-name-is-earl-shibe-and-I-can-paint-that-card-for-$29.95, or whatever) is what we use in when we refer to the specific product.

**Pull request closed without merging**